sdk/download.go

```diff
@@ -269,11 +269,12 @@ func UpdateFile(file, url string, client *http.Client) error {
 	}
 	t.Close()
 	tempFile := t.Name()
-	defer os.Remove(tempFile)
+	os.Remove(tempFile) // let DownloadFile create it and use umask
```

Comment:
Yeah, this completely defeats the use of TempFile. Since I don't think there is any security issue here strict use of TempFile isn't required so there are three options:

 - Always download to some pre-defined but alternate name, just so nothing confuses incomplete files with valid ones. e.g. use `os.Create(file + ".part")`
 - Use some random string, effectively what this this change does but without the pretense of using TempFile
 - Use chmod.

I've got a feeling I've probably done this TempFile+Remove thing before but per-euank's point lets avoid that pattern. :)